Forms: Update save to new format

enejb opened the pull request:
This Pr Implements the saving of the feedback forms in a new format. 

## Proposed changes:
* Update the process_submission function to the the new feedback class and save the submission in the new format. 

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Create a new form with all the fields. 
Save it. 
Can you see the response? Can you export it as before. 
Use the non Ajax way of submitting the form. 
Does it still work as expected? 
Do the values all still match that we are passing into different filters and actions in the process_submission ? 

Do the test pass?

enejb committed:
Forms: Refactor contact form submission to use Feedback object

Refactored the contact form submission process to delegate data extraction, Akismet variable preparation, and entry value handling to the Feedback object. Updated related tests to assert against Feedback methods instead of raw post content. Improved file upload field handling in the dashboard inbox and fixed a bug in JSON decoding for file data. This change centralizes feedback data logic, improves maintainability, and enhances test reliability.

enejb committed:
changelog

simison commented on projects/packages/forms/src/contact-form/class-contact-form.php:
```diff
@@ -1570,6 +1570,13 @@ public function get_field_ids() {
 	 */
 	public function process_submission() {
+		$page_number = 1;
+		if ( isset( $_POST['page'] ) ) { // phpcs:Ignore WordPress.Security.NonceVerification.Missing
+			$page_number = absint( wp_unslash( $_POST['page'] ) ); // phpcs:Ignore WordPress.Security.NonceVerification.Missing
+		}
+
+		$response = Feedback::from_submission( $_POST, $this, $this->current_post, $page_number ); // phpcs:Ignore WordPress.Security.NonceVerification.Missing
```
You could document here why the ignore-rules.

enejb committed:
Revert the save function so that we can

enejb committed:
Update class-contact-form.php

enejb committed:
minor fixes

enejb committed:
revert the test setup

dhasilva reviewed (changes requested):
It is not working for the file upload field:

| Before | After |
| - | - |
| <img width="797" height="147" alt="Screenshot from 2025-08-20 12-34-12" src="https://github.com/user-attachments/assets/88b940c8-28aa-4bb9-bd90-afb65d1d043f" /> | <img width="797" height="147" alt="Screenshot from 2025-08-20 12-34-17" src="https://github.com/user-attachments/assets/c268e7d4-2e4b-45fb-b01c-9fe84894d798" />

enejb committed:
Forms: Add submit context to feedback value rendering

Introduces a 'submit' context to get_all_values and get_legacy_extra_values methods in Feedback, and updates Contact_Form to use this context when retrieving values. Feedback_Field now supports rendering values specifically for submission, with special handling for file fields to ensure only relevant file data is included.

enejb committed:
Remove l function

enejb committed:
Add test for the 'submit' case

enejb committed:
minor

dhasilva reviewed (approved):
Testing fine now!
